src/Microsoft.DotNet.XHarness.CLI/Commands/Android/AndroidRunCommand.cs

```diff
@@ -0,0 +1,261 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.DotNet.XHarness.Android;
+using Microsoft.DotNet.XHarness.Android.Execution;
+using Microsoft.DotNet.XHarness.CLI.CommandArguments.Android;
+using Microsoft.DotNet.XHarness.Common.CLI;
+using Microsoft.DotNet.XHarness.Common.CLI.CommandArguments;
+using Microsoft.DotNet.XHarness.Common.CLI.Commands;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DotNet.XHarness.CLI.Commands.Android
+{
+    internal class AndroidRunCommand : XHarnessCommand
+    {
+        // nunit2 one should go away eventually
+        private static readonly string[] s_xmlOutputVariableNames = { "nunit2-results-path", "test-results-path" };
+        private const string TestRunSummaryVariableName = "test-execution-summary";
+        private const string ShortMessageVariableName = "shortMsg";
+        private const string ReturnCodeVariableName = "return-code";
+        private const string ProcessCrashedShortMessage = "Process crashed";
+
+        private readonly AndroidRunCommandArguments _arguments = new AndroidRunCommandArguments();
+
+        protected override XHarnessCommandArguments Arguments => _arguments;
+
+        protected override string CommandUsage { get; } = "android run --output-directory=... --package-name=... [OPTIONS]";
+
+        private const string CommandHelp = "Run tests using an already installed .apk on an Android device";
+        protected override string CommandDescription { get; } = @$"
+{CommandHelp}
+
+APKs can communicate status back to XHarness using the parameters:
+
+Required:
+{ReturnCodeVariableName} - Exit code for instrumentation. Necessary because a crashing instrumentation may be indistinguishable from a passing one from exit codes.
```

Comment:
```suggestion
{ReturnCodeVariableName} - Exit code for instrumentation. Necessary because a crashing instrumentation may be indistinguishable from a passing one based solely on the exit code.
```